app/models/npc.rb

```diff
@@ -0,0 +1,51 @@
+class Npc < ActiveRecord::Base
+  class ConsequenceLimit < ActiveModel::Validator
+    def validate(npc)
+      if npc.consequences.length > 4
+        npc.errors[:consequence] << 'Cannot have more than 4 consequences'
+      end
+    end
+  end
+
+  validates :name, :npc_type, :system_id, presence: true
+  validates_uniqueness_of :name, scope: :system
+  validates :max_physical_stress, :max_mental_stress, inclusion: 1..5
+  validates_with ConsequenceLimit
+  belongs_to :system
+  has_many :character_skills, dependent: :destroy
+  has_many :stunts, dependent: :destroy
+  has_many :aspects, as: :aspectable
+  has_many :consequences, before_add: :consequence_limit_validation
+
+  def consequence_limit
+    ConsequenceLimit.new
```

Comment:
That's correct. Since you're returning an object instance, I would presume that if I received that instance that it would continue to be available. Now, if I call this method 2 times in a row I will get 2 instances of ConsequenceLimit. `@__cl` is just a name I used for shorthand. I would more than likely give it a name that matched the method name `@consequence_limit ||= ConsequenceLimit.new`